### PR TITLE
HCS-1805: Add support for new cloud consul service version

### PR DESCRIPTION
**Pull request by riddhi89**
JIRA: https://hashicorp.atlassian.net/browse/HCS-1805
Updates the SDK to support the new cloud consul service version.
More details about the new version can be found in https://hashicorp.atlassian.net/browse/HCS-1704

### CODE REVIEW

**Review by Kedevlin — APPROVED** [review]
LGTM! I think @bcmdarroch or @roaks3 (code owners) should take a look for final approval

**Review by roaks3 — APPROVED** [review]
Tested the new client locally and looks good!

**Pull request merged**
